pydatastructs/linear_data_structures/arrays.py

```diff
@@ -139,6 +140,123 @@ def __len__(self):
+    >>> arr = MDA(int, 5, 6, 9)
+    >>> arr.fill(32)
+    >>> arr[3, 0, 0]
+    32
+    >>> arr[3, 0, 0] = 7
+    >>> arr[3, 0, 0]
+    7
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Array_data_structure#Multidimensional_arrays
+
+    """
+    __slots__ = ['_shape', '_data', '_dtype']
+
+    def __new__(cls, dtype=NoneType, *args, **kwargs):
+        if dtype is NoneType or not args:
+            raise ValueError("Array cannot be created due to incorrect"
+                             " information.")
+        if len(args) == 1:
+            obj = Array.__new__(cls)
+            obj._dtype = dtype
+            obj._shape = [1]
+            obj._data = [None] * args[0]
+            return obj
+
+
+        dimensions = args
+        for dimension in dimensions:
+            if dimension < 1:
+                raise ValueError("Array cannot be created due to incorrect"
+                                 " dimensions.")
+        n_dimensions = len(dimensions)
+        d_sizes = []
+        index = 0
+        while n_dimensions > 1:
+            size = dimensions[index]
+            for i in range(index,  len(dimensions) - 1):
+                size = size * dimension
+            d_sizes.append(size)
+            n_dimensions -= 1
+            index += 1
+        d_sizes.append(dimensions[index])
+        d_sizes.append(1)
+        obj = Array.__new__(cls)
+        obj._dtype = dtype
+        obj._shape = d_sizes
+        obj._data = [None] * obj._shape[1] * dimensions[0]
+        return obj
+
+    @classmethod
+    def methods(cls):
+        return ['__new__', '__getitem__', '__setitem__', 'fill']
+
+    def __getitem__(self, indices):
+        self._compare_shape(indices)
+        if isinstance(indices, int):
+            return self._data[indices]
+        position = 0
+        for i in range(0, len(indices)):
+            position += self._shape[i+1] * indices[i]
+        return self._data[position]
+
+    def __setitem__(self, indices, element):
+        self._compare_shape(indices)
+        if isinstance(indices, int):
+            self._data[indices] = element
+        else:
+            position = 0
+            for i in range(0, len(indices)):
+                position += self._shape[i+1] * indices[i]
+            self._data[position] = element
+
+    def _compare_shape(self, indices):
+        if isinstance(indices, int):
+            if len(self._shape) > 1:
+                raise IndexError("Shape mismatch, current shape is %s" % self._shape)
+            return
+        if len(indices) != len(self._shape) - 1:
+            raise IndexError("Shape mismatch, current shape is %s" % self._shape)
+        for i in range(len(indices)):
+            if indices[i] > self._shape[i]:
+                raise IndexError("Index out of range.")
+
+    def fill(self, element):
+        element = self._dtype(element)
+        for i in range(len(self._data)):
+            self._data[i] = element
+
```

Comment:
```suggestion
    def fill(self, element):
        element = self._dtype(element)
        for i in range(len(self._data)):
            self._data[i] = element

    @property
    def shape(self):
        return  self._shape
```
Please test it as well.

Reply:
Add `__str__` method as well. See, ODA.__str__.